client/lib/wp/sync-handler/cache-index.js

```diff
@@ -84,13 +88,13 @@ export const cacheIndex = {
 
 	dropOlderThan( lifetime ) {
 		const dropElders = records => {
-			const droppedRecords = filter( records, rec => {
+			const removeRecords = filter( records, rec => {
```

Comment:
Minor optimization, but we sort these in a single pass for removed vs retained. Rather than a filter, then a difference.